app/src/main/java/org/brightmindenrichment/street_care/ui/visit/details/VisitLogDetailsViewModel.kt

```diff
@@ -55,11 +57,13 @@ class VisitLogDetailsViewModel : ViewModel() {
         viewModelScope.launch {
             try {
                 _visitLog.value?.id?.let { id ->
-                    firestore.collection("VisitLogBook").document(id).delete().await()
-                    _deleteResult.value = true
+                    withContext(Dispatchers.IO) {
+                        firestore.collection("VisitLogBook").document(id).delete().await()
+                    }
+                    _deleteResult.postValue(true)
```

Comment:
This should still be `_deleteResult.value = true` and in the catch block `_deleteResult.value = false` it would be `postValue` if executing still within the background Dispatcher.IO but since that call suspends the UI can be updated with `_deleteResult.value = true`